### PR TITLE
New EFX algorithm - SquareTrue

### PR DESCRIPTION
**Pull request by jhornsby**
Added a new EFX algorithm - SquareTrue with 4 points.  This is my 1st attempt to contribute to QLC+ code - I don't know what to do with the unit tests & make test doesn't work on my build environment (windows).  Hope this is of some use ;-)

### CODE REVIEW

File: engine/src/efx.cpp [hunk 1]
```diff
@@ -381,6 +387,29 @@ void EFX::calculatePoint(float iterator, float* x, float* y) const
         *x = round(cos(iterator));
         *y = round(sin(iterator));
         break;
+		
+	case SquareTrue:
+         if (iterator < M_PI / 2)
```
Comment (mcallegari) [inline engine/src/efx.cpp:392]: Please fix indentation

File: engine/src/efx.h [hunk 2]
```diff
@@ -64,9 +64,11 @@ class Fixture;
 #define KXMLQLCEFXDiamondAlgorithmName      QString("Diamond")
 #define KXMLQLCEFXSquareAlgorithmName       QString("Square")
 #define KXMLQLCEFXSquareChoppyAlgorithmName QString("SquareChoppy")
+#define KXMLQLCEFXSquareTrueAlgorithmName  QString("SquareTrue")
```
Comment (mcallegari) [inline engine/src/efx.h:67]: Please fix indentation

File: engine/src/efx.h [hunk 3]
```diff
@@ -64,9 +64,11 @@ class Fixture;
...
 #define KXMLQLCEFXLeafAlgorithmName         QString("Leaf")
 #define KXMLQLCEFXLissajousAlgorithmName    QString("Lissajous")
 
+
```
Comment (mcallegari) [inline engine/src/efx.h:71]: Extra line not needed

**Review by mcallegari — CHANGES_REQUESTED** [review]

File: engine/src/efx.h (hunk 2 above)
Comment (jhornsby) [reply engine/src/efx.h]: Fixed in 5a3d1edc46f3b7e7601b6d34ee5964406f30adc4  (sorry!)

**Comment by mcallegari** [comment]
I'm gonna merge this and deal with test units myself.
Thanks for the contribution :+1:

**Pull request merged**